### PR TITLE
New package: Parsel

### PR DESCRIPTION
**Pull request by frerappa**
Parsel is a BSD-licensed Python library to extract and remove data from HTML and XML using XPath and CSS selectors, optionally combined with regular expressions.
@hyviquel

### CODE REVIEW

File: var/spack/repos/builtin/packages/py-parsel/package.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    url = "https://github.com/scrapy/parsel/archive/refs/tags/v1.6.0.tar.gz"
+
+    maintainers = ["frerappa"]
+
+    version("1.6.0", sha256="15546bb637074725e4b2c0d7c579356636251389d9b296520543d835d615e83d")
+
+    depends_on("py-cssselect", type="build")
+    depends_on("py-lxml", type="build")
+    depends_on("py-six", type="build")
+    depends_on("py-w3lib", type="build")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-parsel/package.py:24]: Missing build deps on setuptools and pytest-runner

Missing dependency versions:
```python
install_requires = [                                                                     
    'w3lib>=1.19.0',                                                                     
    'lxml',                                                                              
    'six>=1.6.0',                                                                        
    'cssselect>=0.9'                                                                     
]                                                                                        
extras_require = {}                                                                      
                                                                                         
if not has_environment_marker_platform_impl_support():                                   
    if sys.version_info[0:2] < (3, 0):                                                   
        install_requires.append("functools32")                                           
else:                                                                                    
    extras_require[":python_version<'3.0'"] = ["functools32"]   
```
`install_requires` should be `type=("build", "run")`

File: var/spack/repos/builtin/packages/py-w3lib/package.py [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+class PyW3lib(PythonPackage):
+    """This is a Python library of web-related functions, such as:
+    - remove comments, or tags from HTML snippets
+    - extract base url from HTML snippets
+    - translate entites on HTML strings
+    - convert raw HTTP headers to dicts and vice-versa
+    - construct HTTP auth header
+    - converting HTML pages to unicode
+    - sanitize urls (like browsers do)
+    - extract arguments from urls
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-w3lib/package.py:18]: When you run `spack info`, these newline characters are removed, so this won't be very legible. I would remove everything except the first line.

File: var/spack/repos/builtin/packages/py-parsel/package.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+from spack.package import *
+
+
+class PyParsel(PythonPackage):
+    """Parsel is a BSD-licensed Python library to extract and remove data from
+    HTML and XML using XPath and CSS selectors and regular expressions.
+    """
+
+    homepage = "https://parsel.readthedocs.io"
+    url = "https://github.com/scrapy/parsel/archive/refs/tags/v1.6.0.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-parsel/package.py:15]: https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#pypi-vs-github

File: var/spack/repos/builtin/packages/py-w3lib/package.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    """
+
+    homepage = "https://github.com/scrapy/w3lib"
+    url = "https://github.com/scrapy/w3lib/archive/refs/tags/v1.22.0.tar.gz"
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-w3lib/package.py:22]: https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html#pypi-vs-github

File: var/spack/repos/builtin/packages/py-w3lib/package.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+    maintainers = ["frerappa"]
+
+    version("1.22.0", sha256="d388f06a11008c68f96c95224e224691e5ce9c13733d89bc85a550c448f83c97")
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/py-w3lib/package.py:26]: Missing build dep on setuptools

Missing build/run dep on six:
```python
    install_requires=['six >= 1.4.1'],                                                   
```

**Review by adamjstewart — CHANGES_REQUESTED** [review]
If this is your first time packaging Python libraries, I recommend reading through https://spack.readthedocs.io/en/latest/build_systems/pythonpackage.html

**Comment by bernhardkaindl** [comment]
@frerappa Would you apply the changes from the review?

**Comment by bernhardkaindl** [comment]
> @frerappa Would you apply the changes from the review?

@frerappa ping?

**Comment by alalazo** [comment]
Closing, since there has been no activity in a while. Feel free to reopen if you want to continue work on this.

**Pull request closed without merging**